cluster-autoscaler/cloudprovider/gce/gce_manager.go

```diff
@@ -283,11 +283,12 @@ func (m *gceManagerImpl) GetMigForInstance(instance GceRef) (Mig, error) {
 
 // GetMigNodes returns mig nodes.
 func (m *gceManagerImpl) GetMigNodes(mig Mig) ([]cloudprovider.Instance, error) {
-	return m.GceService.FetchMigInstances(mig.GceRef())
+	return m.migInfoProvider.GetMigInstances(mig.GceRef())
```

Comment:
That makes a lot of sense, but the caching slightly changes the contract for this method (getting fresh vs cached list). I think it's fine for the current CA implementation, as long as the cache is refreshed each loop (as opposed to, for example, in forceRefresh()). Your implementation is correct in that sense, but you need to be increasingly aware what is cached for how long. 
Maybe consider adding a comment somewhere (not sure what is the best place) about expected lifetime of all the caches? WDYT?

Reply:
Yeah, I think a comment above the `GceCache` would be a good place for that info. There are some other caches that are dependant on the same logic - base names, target sizes and instance template names are also maintained in cache and refreshed after each loop.